pkg/assembler/assembler.go

```diff
@@ -151,48 +151,48 @@ func (g *Graph) AppendGraph(gs ...Graph) {
 // ingestion of the software trees are implicit and handled by the
 // client library.
 type IngestPredicates struct {
-	CertifyScorecard []CertifyScorecardIngest
-	IsDependency     []IsDependencyIngest
-	IsOccurrence     []IsOccurrenceIngest
-	HasSlsa          []HasSlsaIngest
-	CertifyVuln      []CertifyVulnIngest
-	IsVuln           []IsVulnIngest
-	HasSourceAt      []HasSourceAtIngest
-	CertifyBad       []CertifyBadIngest
-	CertifyGood      []CertifyGoodIngest
-	HasSBOM          []HasSBOMIngest
-	HashEqual        []HashEqualIngest
-	PkgEqual         []PkgEqualIngest
-	Vex              []VexIngest
+	CertifyScorecard []CertifyScorecardIngest `json:"certifyScorecard,omitempty"`
+	IsDependency     []IsDependencyIngest     `json:"isDepedency,omitempty"`
```

Comment:
question on this, wouldn't this allow arbitrary data to be added to the database without being tied to an originating document? Seems like we have to limit access to this. I can see this being useful for certifyBad and CertifyGood but for the others, we should be careful.

Reply:
Seems like this should be more of an "internal" use only. Not sure how we enforce that.

Reply:
We can make the new parser hidden behind a flag, I think.

It will be hard to block this on GQL layer, as by that time it is not too possible to differentiate between ingestion requests coming from this usecase or from the normal flow.

Reply:
one thing we can do is highlight it via the naming scheme. it won't prevent it, but at least make it harder for folks to claim otherwise. Or hide some of it behind an internal api so even if folks mess with it, it should be crystal clear that they're doing something unsafe.

e.g. https://hackage.haskell.org/package/bytestring-0.11.4.0/docs/Data-ByteString-Internal.html#v:accursedUnutterablePerformIO

Reply:
yea i think ill remove this from the default ingestor and then have a separate `guacone danger collect` or something similar